src/index.md

```diff
@@ -14,15 +14,40 @@ in the officially maintained [Rust Book](https://doc.rust-lang.org/book/).
 To read the book about gdnative (Godot 3 binding), follow [this link](../gdnative-book).
 
 
+## The basics
+
+A **library** is in its most basic form is a **collection of functions**. Godot has several of its own built-in libraries, but like most other programs made as a creation tool, it also has the capacity to access collections of libraries made outside of Godot. Those are called **extensions**, and like most other software, they usually come in separate files -- typically in the form of file extensions such as `.so` if you're on Linux, `.dll` if you're on Windows, or `.dylib` on MacOS.
+
+The way that GDExtension work is that you basically expose these files to your Godot project by using a `.gdextension` file pointing to the thing. Such file will have all the information of where that file is and which version of Godot it should be compatible with.
+
+The job of a binding library is that we go the other way and we instead treat Godot itself like a library, so that we can have most of the interactions of the engine happen in our language of choice. In practice,  Godot is still going to be the host engine of our game, so we have to expose things back to Godot so that it can use and make available in the editor -- think of how you use the [`@export`](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_exports.html) flags to expose bits and pieces of a Node to the editor UI with shiny buttons. that's what we're gonna do but much deeper.
+
+```mermaid
+flowchart TD
+   gdp["Your Project in Godot"]
+   --the editor looks for extensions to include-->
+   gdext[".gdextension file with pointers to the library file"]
+   -.Uses paths or symbolic links to reach your compiled library.->
+   lib["libyour-project.so \nlibyour-project.dll \nlibyour-project.dylib"]
+   --"the library is compiled into\n a readable format from the compiler"----
+   proj["Your own library written in Rust"]
+
+   proj ==your library makes nodes,\n components and another shenanigans\n available for Godot to use==> gdp
+
+```
+
+
+# About the extension
+
 ## Terminology
 
-To avoid confusion, here is an explanation of names and technologies you may encounter over time.
+To avoid confusion, here is an explanation of names and technologies you may encounter over time. This is what we will be referring to at the mention of the terms listed from here now on:
```

Comment:
I don't see what the 2nd sentence adds. I would avoid unnecessary repetition to decrease reading time.